host/src/main/java/com/instana/dc/host/AbstractHostDc.java

```diff
@@ -76,11 +95,45 @@ public void initMeters(OpenTelemetry openTelemetry) {
         initMeter(openTelemetry, HostDcUtil.MeterName.FILESYSTEM);
         initMeter(openTelemetry, HostDcUtil.MeterName.PROCESSES);
         initMeter(openTelemetry, HostDcUtil.MeterName.PAGING);
+        initMeter(openTelemetry, HostDcUtil.MeterName.IBMQMGR);
     }
 
+    private void addShutdownHook() {
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            closeProcess();
+        }));
+    }
+
+    private void closeProcess() {
+        try {
+            if (bufferedReader != null) {
+                bufferedReader.close();
+            }
+        } catch (IOException e) {
+            logger.severe("Cannot close the bufferedReader: " + e.getMessage());
+        }
+        if (process != null) {
+            process.destroy();
+        }
+        if (exec != null && !exec.isShutdown()) {
+            exec.shutdownNow();
+        }
+    }
 
     @Override
     public void start() {
-        exec.scheduleWithFixedDelay(this::collectData, 1, pollInterval, TimeUnit.SECONDS);
+        if (hostDcType.toUpperCase().equals("MQ_APPLIANCE")){
```

Comment:
Those code should not be in the AbstraceHostDc class because other host DC implementation does not need it.